dask_kubernetes/operator/controller/controller.py

```diff
@@ -481,6 +481,29 @@ async def get_desired_workers(scheduler_service_name, namespace, logger):
 worker_group_scale_locks = defaultdict(lambda: asyncio.Lock())
 
 
+@kopf.on.field("daskcluster.kubernetes.dask.org", field="spec.worker.replicas")
+async def daskcluster_default_worker_group_replica_update(
+    name, namespace, meta, spec, old, new, body, logger, **kwargs
+):
+    if old is None:
+        return
+    worker_group_name = f"{name}-default"
+
+    async with kubernetes.client.api_client.ApiClient() as api_client:
+        custom_objects_api = kubernetes.client.CustomObjectsApi(api_client)
+        custom_objects_api.api_client.set_default_header(
+            "content-type", "application/merge-patch+json"
+        )
+        await custom_objects_api.patch_namespaced_custom_object_scale(
+            group="kubernetes.dask.org",
+            version="v1",
+            plural="daskworkergroups",
+            namespace=namespace,
+            name=worker_group_name,
+            body={"spec": {"replicas": new}},
+        )
```

Comment:
I'm excited about the `kr8s` changes in #696 because this will be hugely simplified.

```suggestion
    worker_group = await DaskWorkerGroup.get(worker_group_name)
    await worker_group.scale(new)
```

Happy to merge this as-is though and I'll update it in a follow-up once `kr8s` is integrated nicely.

Reply:
Oh, wow! Definitely looking forward to that!